### ISSUE TITLE
Unable to see any content in the vCenter VIC Plugin UI 

### ISSUE THREAD

**Issue by codyde** [vmware/vic-ui#299]
**User Statement:**
With a fresh install of the VIC plugin 1.3, I am unable to see anything in the plugin interface. 

**Details:**

I installed VIC 1.3 and the plugin fresh in my environment. I am currently running vCenter 6.5 u1d. This environment started off as a 6.5 GA environment, and has been subsequently updated throughout the major releases. I believe my pattern was 6.5 GA to 6.5d, to 6.5u1, to 6.5 u1d. 

There are a number of untrusted certificate errors in the "ERROR" logs (pasted below). For context, i'm running the vCenter Certificate Authority with my domain controllers root certificate as the trusted root - the VMCA is an intermediate certificate authority. I'm not certain if this has any bearing on the problem but I wanted to specify it. 

I have removed and reinstalled the client, stopped services individually and restarted, rebooted the apliance, and upgraded it from 6.5u1 to 6.5u1d to test resolutions - unsuccessful on all accounts. 

 
**VIC version:**

1.3

**Deployment details:**

N/A - VIC vCenter UI Plugin Related 

**Logs:**
Grep of vsphere_client_virgo.log with "com.vmware.vic" - 

root@hlcorevc01 [ /var/log/vmware/vsphere-ui/logs ]# cat vsphere_client_virgo.log | grep "com.vmware.vic"
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86)
        at com.vmware.vic.PropFetcher.getServiceContent(PropFetcher.java:706)
        at com.vmware.vic.PropFetcher.getVicVms(PropFetcher.java:198)
        at com.vmware.vic.ObjectStore.getRootObject(ObjectStore.java:96)
        at com.vmware.vic.ObjectStore.getObj(ObjectStore.java:125)
        at com.vmware.vic.VicUIDataAdapter.processSimpleConstraint(VicUIDataAdapter.java:189)
        at com.vmware.vic.VicUIDataAdapter.processConstraint(VicUIDataAdapter.java:135)
        at com.vmware.vic.VicUIDataAdapter.processQuerySpec(VicUIDataAdapter.java:106)
        at com.vmware.vic.VicUIDataAdapter.getData(VicUIDataAdapter.java:86...

GREP of vsphere_client_virgo.log with "ERROR" 

root@hlcorevc01 [ /var/log/vmware/vsphere-ui/logs ]# cat vsphere_client_virgo.log | grep "ERROR"
[2018-01-10T06:14:05.093Z] [ERROR] http-bio-5090-exec-8          o.a.c.core.ContainerBase.[Catalina].[localhost].[/ui].[spring]    Servlet.service() for servlet [spring] in context with path [/ui] threw exception [Request processing failed; nested exception is com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match] with root cause java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:06.026Z] [ERROR] http-bio-5090-exec-8          o.a.c.core.ContainerBase.[Catalina].[localhost].[/ui].[jsp]       Servlet.service() for servlet jsp threw exception java.lang.IllegalStateException: getOutputStream() has already been called for this response
[2018-01-10T06:14:05.076Z] [ERROR] http-bio-5090-exec-11         org.apache.catalina.core.ContainerBase.[Catalina].[localhost]     Exception Processing ErrorPage[errorCode=0, location=/error.jsp] org.apache.jasper.JasperException: java.lang.IllegalStateException: getOutputStream() has already been called for this response
[2018-01-10T06:14:06.029Z] [ERROR] http-bio-5090-exec-8          org.apache.catalina.core.ContainerBase.[Catalina].[localhost]     Exception Processing ErrorPage[errorCode=0, location=/error.jsp] org.apache.jasper.JasperException: java.lang.IllegalStateException: getOutputStream() has already been called for this response
[2018-01-10T06:14:31.648Z] [ERROR] data-service-pool-174        70000119 100005 200001 com.vmware.utils.ssl.ThumbprintTrustManager                       Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:31.673Z] [ERROR] data-service-pool-176        70000121 100005 200001 com.vmware.utils.ssl.ThumbprintTrustManager                       Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:31.710Z] [ERROR] http-bio-5090-exec-2         70000121 100005 200001 com.vmware.vise.mvc.servlets.GlobalExceptionResolver              Exception handled:  com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:31.726Z] [ERROR] http-bio-5090-exec-13        70000119 100005 200001 com.vmware.vise.mvc.servlets.GlobalExceptionResolver              Exception handled:  com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:31.728Z] [ERROR] http-bio-5090-exec-2          o.a.c.core.ContainerBase.[Catalina].[localhost].[/ui].[spring]    Servlet.service() for servlet [spring] in context with path [/ui] threw exception [Request processing failed; nested exception is com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match] with root cause java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:31.734Z] [ERROR] http-bio-5090-exec-13         o.a.c.core.ContainerBase.[Catalina].[localhost].[/ui].[spring]    Servlet.service() for servlet [spring] in context with path [/ui] threw exception [Request processing failed; nested exception is com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match] with root cause java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:31.731Z] [ERROR] http-bio-5090-exec-2          o.a.c.core.ContainerBase.[Catalina].[localhost].[/ui].[jsp]       Servlet.service() for servlet jsp threw exception java.lang.IllegalStateException: getOutputStream() has already been called for this response
[2018-01-10T06:14:31.737Z] [ERROR] http-bio-5090-exec-2          org.apache.catalina.core.ContainerBase.[Catalina].[localhost]     Exception Processing ErrorPage[errorCode=0, location=/error.jsp] org.apache.jasper.JasperException: java.lang.IllegalStateException: getOutputStream() has already been called for this response
[2018-01-10T06:14:31.755Z] [ERROR] http-bio-5090-exec-13         o.a.c.core.ContainerBase.[Catalina].[localhost].[/ui].[jsp]       Servlet.service() for servlet jsp threw exception java.lang.IllegalStateException: getOutputStream() has already been called for this response
[2018-01-10T06:14:31.757Z] [ERROR] http-bio-5090-exec-13         org.apache.catalina.core.ContainerBase.[Catalina].[localhost]     Exception Processing ErrorPage[errorCode=0, location=/error.jsp] org.apache.jasper.JasperException: java.lang.IllegalStateException: getOutputStream() has already been called for this response
[2018-01-10T06:14:31.808Z] [ERROR] data-service-pool-181        70000123 100005 200001 com.vmware.utils.ssl.ThumbprintTrustManager                       Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:31.816Z] [ERROR] data-service-pool-183        70000124 100005 200001 com.vmware.utils.ssl.ThumbprintTrustManager                       Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:31.871Z] [ERROR] http-bio-5090-exec-11        70000123 100005 200001 com.vmware.vise.mvc.servlets.GlobalExceptionResolver              Exception handled:  com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:31.873Z] [ERROR] http-bio-5090-exec-8         70000124 100005 200001 com.vmware.vise.mvc.servlets.GlobalExceptionResolver              Exception handled:  com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:31.884Z] [ERROR] http-bio-5090-exec-8          o.a.c.core.ContainerBase.[Catalina].[localhost].[/ui].[spring]    Servlet.service() for servlet [spring] in context with path [/ui] threw exception [Request processing failed; nested exception is com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match] with root cause java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:31.886Z] [ERROR] http-bio-5090-exec-8          o.a.c.core.ContainerBase.[Catalina].[localhost].[/ui].[jsp]       Servlet.service() for servlet jsp threw exception java.lang.IllegalStateException: getOutputStream() has already been called for this response
[2018-01-10T06:14:31.888Z] [ERROR] http-bio-5090-exec-11         o.a.c.core.ContainerBase.[Catalina].[localhost].[/ui].[spring]    Servlet.service() for servlet [spring] in context with path [/ui] threw exception [Request processing failed; nested exception is com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match] with root cause java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:31.894Z] [ERROR] http-bio-5090-exec-11         o.a.c.core.ContainerBase.[Catalina].[localhost].[/ui].[jsp]       Servlet.service() for servlet jsp threw exception java.lang.IllegalStateException: getOutputStream() has already been called for this response
[2018-01-10T06:14:31.896Z] [ERROR] http-bio-5090-exec-8          org.apache.catalina.core.ContainerBase.[Catalina].[localhost]     Exception Processing ErrorPage[errorCode=0, location=/error.jsp] org.apache.jasper.JasperException: java.lang.IllegalStateException: getOutputStream() has already been called for this response
[2018-01-10T06:14:31.897Z] [ERROR] http-bio-5090-exec-11         org.apache.catalina.core.ContainerBase.[Catalina].[localhost]     Exception Processing ErrorPage[errorCode=0, location=/error.jsp] org.apache.jasper.JasperException: java.lang.IllegalStateException: getOutputStream() has already been called for this response
[2018-01-10T06:14:31.971Z] [ERROR] data-service-pool-193        70000126 100005 200001 com.vmware.utils.ssl.ThumbprintTrustManager                       Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:32.080Z] [ERROR] data-service-pool-195        70000128 100005 200001 com.vmware.utils.ssl.ThumbprintTrustManager                       Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:32.110Z] [ERROR] http-bio-5090-exec-14        70000126 100005 200001 com.vmware.vise.mvc.servlets.GlobalExceptionResolver              Exception handled:  com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:32.119Z] [ERROR] http-bio-5090-exec-14         o.a.c.core.ContainerBase.[Catalina].[localhost].[/ui].[spring]    Servlet.service() for servlet [spring] in context with path [/ui] threw exception [Request processing failed; nested exception is com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match] with root cause java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:32.123Z] [ERROR] http-bio-5090-exec-14         o.a.c.core.ContainerBase.[Catalina].[localhost].[/ui].[jsp]       Servlet.service() for servlet jsp threw exception java.lang.IllegalStateException: getOutputStream() has already been called for this response
[2018-01-10T06:14:32.156Z] [ERROR] http-bio-5090-exec-14         org.apache.catalina.core.ContainerBase.[Catalina].[localhost]     Exception Processing ErrorPage[errorCode=0, location=/error.jsp] org.apache.jasper.JasperException: java.lang.IllegalStateException: getOutputStream() has already been called for this response
[2018-01-10T06:14:32.157Z] [ERROR] data-service-pool-197        70000130 100005 200001 com.vmware.utils.ssl.ThumbprintTrustManager                       Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:32.165Z] [ERROR] http-bio-5090-exec-12        70000128 100005 200001 com.vmware.vise.mvc.servlets.GlobalExceptionResolver              Exception handled:  com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:32.191Z] [ERROR] http-bio-5090-exec-12         o.a.c.core.ContainerBase.[Catalina].[localhost].[/ui].[spring]    Servlet.service() for servlet [spring] in context with path [/ui] threw exception [Request processing failed; nested exception is com.sun.xml.internal.ws.client.ClientTransportException: HTTP transport error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match] with root cause java.security.cert.CertificateException: Server certificate chain is not trusted and thumbprint doesn't match
[2018-01-10T06:14:32.193Z] [ERROR] http-bio-5090-exec-12         o.a.c.core.ContainerBase.[Catalina].[localhost].[/ui].[jsp]       Servlet.service() for servlet jsp threw exception java.lang.IllegalStateException: getOutputStream() has already been called for this response
[2018-01-10T06:14:32.202Z] [ERROR] http-bio-5090-exec-12         org.apache.catalina.core.ContainerBase.[Catalina].[localhost]     Exception Processing ErrorPage[errorCode=0, location=/error.jsp] org.apache.jasper.JasperException: java.lang.IllegalStateEx...

**Comment by jooskim** [comment]
After a Slack conversation with Cody this issue seems to be coming from a previous issue on SSL certificate thumbprint not matching between vCenter Server PSC and the browser.

**Comment by jooskim** [comment]
Same issue reported here: https://github.com/vmware/vic-product/issues/1309

**Comment by jooskim** [comment]
Some investigations with Cody revealed that two certificates were being verified against the ThumbprintTrustManager's thumbprints store where it contains only the VC appliance's certificate. The other thumbprint that was missing in the store was coming from the external PSC.
Until now I could not reproduce the problem because all our test environments were based on an embedded. Fix should not be too challenging but verifying the fix would be non-trivial where we have to first deploy a VCSA 6.5 with an external PSC and then upgrade it to a newer 6.5 build. At least I finally seem to have found the root cause of this issue.

**Comment by MrVitek** [comment]
@jooskim I am one of the people that have this problem in issue#1309. I am using the embedded psc and still have the problem.

**Comment by jooskim** [comment]
@MrVitek can you elaborate on your VC setup please? For example whether you have upgraded it to a newer version and from an external PSC to an embedded one etc. 

My guess is that until you upgraded your vCSA, the X509 cert had been identical between your VCSA and PSC. And upgrading the VCSA probably caused the cert out of sync.

**Comment by MrVitek** [comment]
@jooskim It has always been an embedded PSC. It has been upgraded since 6.0 if I remember correctly.

**Comment by jooskim** [comment]
@MrVitek Ok then it seems to have something to do with upgrade. Will investigate more and get back to you.

**Comment by MrVitek** [comment]
@jooskim  Here is maybe a weird update, but it all started working out of the blue. The only change made to my enviroment was updating NSX to 6.4 , and the only thing that might be relevant to this(that i can see) is that NSX 6.4 installs an H5 plugin. 

I even did an uninstall and reinstall of VIC and its plugins after this, and the plugin is working fine. Not sure what this tells you but atleast some more information. :)

**Comment by codyde** [comment]
Whoa! Mine too is working now after upgrading to NSX 6.4. Didn't even think
to check! How strange...

**Comment by vmwarelab** [comment]
Well slap me and call me Susan! What the hell.. just doing the NSX upgrade now

**Comment by jooskim** [comment]
@MrVitek @codyde @vmwarelab It's super weird... I would definitely need to replicate your environment later to see what really is going on between NSX and vSphere Client, but I am glad that upgrading NSX to 6.4 fixed the problem. What was the version of NSX your lab instance had?

**Comment by codyde** [comment]
I was running 6.3, upgraded just my manager thus far to 6.4. All
functionality seems solid at this point.

**Comment by vmwarelab** [comment]
@jooskim Hate to break it to ya Josh .. but it didn't fix it in my lab aftger upgrading to NSX 6.4 even after re-installing the plug-in and restarting the both vsphere client

**Comment by vmwarelab** [comment]
@jooskim Soooooo after i clicked on the Network and security in the HTML 5 client .. i dont if it was that or it just needed it time but its working now ..

**Comment by jooskim** [comment]
@vmwarelab Well, I'm happy and frustrated at the same time to hear the plugin is working for you finally. So you just clicked on Networks and it started to work? Did you get to try the new VCH Create wizard?

**Comment by vmwarelab** [comment]
@jooskim Yes Clicked on Network and Security in the HTML 5 Client to access the new NSX HTML 5 plug-in
and yes i did deploy a VCH successfully using the wizard. .. its Awesome!

**Comment by jooskim** [comment]
@vmwarelab Huh, I don't know what just happened with NSX upgrade, but certainly it did seem to work for you :) Any other vSphere Client plugins you have other than NSX and VIC?

**Comment by jooskim** [comment]
After having an in-depth discussion with @dbarkelew , it came to me as a possible root cause that certificates stored in the key store file that vSphere Client uses might have been out of sync with other key stores in the VC appliance that I assume were properly updated by using vecs-cli or certificate manager. I now have a test case that might prove if my guess is right.

1. Deploy a VC
2. Inspect `/etc/vmware/vsphere-ui/keystore.jks` for the vSphere Client
3. Using certificate manager, replace the VMCA root certificate and replace all certificates
4. Install the VIC UI plugin and see if, in the Virgo server log, there is any thumbprint mismatch entry
5. Inspect again `/etc/vmware/vsphere-ui/keystore.jks` and see if it is identical to what vecs-cli reports

**Comment by jooskim** [comment]
I think I finally found the root cause of this issue. When the Root CA on the VCSA is updated (by upgrading to a newer build of vSphere or running VMCA Certificate Manager), its subordinate certificates need to be updated properly. If a certificate fails to be updated for some reason, solutions that utilize the service are likely to be impacted.

In the case of the VIC H5 Client plugin, the issue was with the SSL key vSphere lookup service's certificate not being updated. `vic-service.jar`, which is the Java service for the VIC H5 Client plugin, is dependent on the lookup service, and since its public key didn't match that of the machine SSL cert key, the service failed to make an SSL handshake.

Following are steps I took to investigate this issue:
1. Deploy a VCSA 6.5u1d VM and an ESXi VM on Nimbus
2. Deploy the latest `vic-stage-` prefixed product OVA to the VC
3. Register the appliance through the Getting Started page and install the UI plugin
4. Confirm the functionality of the plugin in the H5 Client
5. Run `/usr/lib/vmware-vmca/bin/certificate-manager` on the VCSA and choose to replace the Root CA
6. Once completed, hit the VC endpoint and log into H5 Client to see if the plugin still works

Interestingly, I could confirm the plugin working consistently even after replacing the Root CA three times in a row. Then I (intentionally) compromised the key store for the vSphere Client and then the plugin stopped working and its dreadful thumbprint mismatching errors started to show up in `vsphere_client_virgo.log`. This confirms that the `vic-service.jar` validates the lookup service certificate using the H5 Client key store. After this I ran certificate manager again to update the Root CA, but maybe I touched something wrong along the way and some services were not properly updated, which essentially led to a state where lookup service had an out-of-sync certificate. For a case like this, manual replacement seems like the only way (@dbarkelew I'd appreciate your thoughts here), and I believe some of you folks @vmwarelab @MrVitek @codyde ran into this state. It is still weird to me that upgrading NSX solved the issue, but it is possible that the certificate issues were corrected during the process.

This is what I have found to be a way to check if the VIC H5 Client Plugin will have a certificate issue in your VC setup. Run the following commands while logged into your VCSA as root and they should generate the exact same content for the SSL public key:

1. `/usr/lib/vmware-vmafd/bin/vecs-cli entry getcert --store MACHINE_SSL_CERT --alias __MACHINE_CERT`
2. `keytool -exportcert -keystore /etc/vmware/vsphere-ui/keystore.jks -v -alias __machine_cert -rfc`
3. `/usr/lib/vmidentity/tools/scripts/lstool.py list --url https://localhost/lookupservice/sdk --no-check-cert --ep-type com.vmware.cis.cs.identity.sso 2>/dev/null`
4. `/usr/lib/vmidentity/tools/scripts/lstool.py list --url https://localhost/lookupservice/sdk --no-check-cert --ep-type com.vmware.vim 2>/dev/null`

If for some reason the key from 3 or 4 doesn't match that of 1 or 2 then the plugin will NOT work, and it signals that there is a certification problem with your VC. How to resolve the issue is beyond my knowledge but I found a relevant link that might be useful: https://kb.vmware.com/s/article/2118939. Again, @dbarkelew I appreciate your inputs on this topic.

**Comment by dbarkelew** [comment]
@jooskim When you say key do you mean the SSL public key? 

For embedded PSC this looks correct. If external step 1 and 2 needs to be done on both VC and external PSC. Then step 3 needs to match results from the external PSC and 4 match results from VC.

Really depend on the overarching situation solutions would be different but in general if 1 and 2 match each other per VC and PSC then this KB this is good. https://kb.vmware.com/kb/2109074 Update the results of ls_tool to match what is in the VECS cert store. 

Now if VECS does not match the jks keystore....that would need further looking.

**Comment by jooskim** [comment]
@dbarkelew Yes I meant public key. I haven't looked into this with a VCSA with an external PSC but it should be future work. How can I update the ls_tool?

**Comment by dbarkelew** [comment]
The KB I linked has two KBs for the steps.
Embedded PSC - https://kb.vmware.com/kb/2121689
External PSC - https://kb.vmware.com/kb/2121701 

In a nutshell get the thumbprint of the old cert in the ls_tool output. Use this to inject the new cert for each entry with the old thumbprint using 
`python ls_update_certs.py --url https://psc.vmware.com/lookupservice/sdk --fingerprint 11:11:AF:D8:CF:27:6B:EF:F7:49:20:3E:D7:90:8C:F6:A0:A2:E2:30 --certfile /certificates/new_machine.crt --user Administrator@vsphere.local --password "Password"`

Careful, do not get the thumbprints for VC endpoints and PSC endpoints mixed. It is very time consuming to reverse to say the least. I recommend snapshotting all VCs and PSCs before attempting this just incase.

**Comment by jooskim** [comment]
I think I am getting closer to documenting solutions to this problem. These are very complicated instructions to follow so I want to ensure with @stuclem and others before moving forward. Stuart, would you please read through the draft and give me your thoughts or questions?

--- start content ---

In case you have installed the plugin, restarted the vSphere Client service and are not still seeing the plugin not working properly, you might be encountering the SSL certificate mismatch issue. To check if you are impacted by the issue, run the following commands to verify if you get the same SSL public key:

Run the following steps while logged into your VCSA as root:

1. (*) Run `/usr/lib/vmware-vmafd/bin/vecs-cli entry getcert --store MACHINE_SSL_CERT --alias __MACHINE_CERT > machine_cert.pem`
2. (*) Run `keytool -exportcert -keystore /etc/vmware/vsphere-ui/keystore.jks -v -alias __machine_cert -rfc > virgo_keystore_machine_cert.pem`
    1. If prompted for password just hit enter
3. Run `/usr/lib/vmidentity/tools/scripts/lstool.py list --url https://localhost/lookupservice/sdk --no-check-cert --ep-type com.vmware.cis.cs.identity.sso 2>/dev/null`
    1. From the output, copy all the text after "SSL trust: " into clipboard
    2. Open a new file (sso_cert.pem) using vim, nano or any editor and Add the following lines:
        1. -----BEGIN CERTIFICATE-----
        2. -----END CERTIFICATE-----
    3. Paste the clipboard content between the two lines you just added and save it
4. Run `/usr/lib/vmidentity/tools/scripts/lstool.py list --url https://localhost/lookupservice/sdk --no-check-cert --ep-type com.vmware.vim 2>/dev/null`
    1. From the output, copy all the text after "SSL trust: " into clipboard
    2. Open a new file (vim25_cert.pem) using vim, nano or any editor and Add the following lines:
        1. -----BEGIN CERTIFICATE-----
        2. -----END CERTIFICATE-----
    3. Paste the clipboard content between the two lines you just added and save it

(*) means if you have an external PSC you need to perform it twice - once in VCSA and the other in PSC.

You should now have 4 different PEM files ready to be compared.
While still logged into VCSA, run the following commands:
1. `openssl x509 -in machine_cert.pem -noout -fingerprint -sha1`
2. `openssl x509 -in virgo_keystore_machine_cert.pem -noout -fingerprint -sha1`
3. `openssl x509 -in sso_cert.pem -fingerprint -sha1`
4. `openssl x509 -in vim25_cert.pem -fingerprint -sha1`

**Embedded PSC setup**
Fingerprint should be identical across all the 4 files

**External PSC setup**
1 and 2 should match each other per VCSA and PSC.
3 should match the PSC fingerprint
4 should match the VCSA fingerprint

If you are seeing different results, it signals that there is a certificate problem with your VC. Resources on how to resolve the certificate issue are as follows:

**vCenter Server or Platform Services Controller certificate validation error for external VMware Solutions in vSphere 6.0 (2109074)**
https://kb.vmware.com/s/article/2109074

**vCenter Server or Platform Services Controller certificate validation error for external VMware Solutions in vSphere 6.0 (2109074)**
https://kb.vmware.com/s/article/2118939

**vCenter Server certificate validation error for external solutions in environments with Embedded Platform Services Controller (2121689)**
https://kb.vmware.com/kb/2121689

vCenter Server or Platform Services Controller certificate validation error messages for external solutions in environments with a External Platform Services Controller (2121701)
https://kb.vmware.com/kb/2121701

--- end content ---

@dbarkelew could you also provide some feedback on this so I can make sure I didn't get critical pieces wrong?

**Comment by stuclem** [comment]
@jooskim I can do a copy edit in your proposed draft directly in the comment above. I'll let you know when I have done. As for how we document this, this looks a lot like a KB article to me. It's definitely too big for the release notes. I can then link to the KB as the workaround in the release notes.

@jooskim @dbarkelew what do you think?

**Comment by stuclem** [comment]
Proposed release note:

- <b>Plug-in does not appear after successful installation and restart of the vSphere Client services. [#299](https://github.com/vmware/vic-ui/issues/299)</b><br>If you have successfully installed the plug-in and restarted the vSphere Client services and you do not see the plug-in in the vSphere Client, you might be encountering an SSL certificate mismatch issue. This can occur if the root CA on the vCenter Server Virtual Appliance is updated, either by upgrading to a newer build of vSphere or by running the VMCA Certificate Manager.

   Workaround: Follow the procedure in the temporary documentation in [Github issue 299](https://github.com/vmware/vic-ui/issues/299#issuecomment-363210223). A formal KB article about this issue is in progress.

**Comment by dbarkelew** [comment]
Issue is because the VECS stores have chain certificates instead of leaf certificates. The VMCA root cert being changed doesn't impact this directly. Still can happen without changing the CA certs and only regenerating the machine_ssl and solution users certificate if things go wrong. 

I do not expect upgrading vCenter to make a difference once the issue exist. Typically will require certificate regeneration or manual correction.

I have not personally fully confirmed that running the VMCA certificate manager tool fixes the issue once impacted. The self signed regeneration worked for me and seems reasonable to me if done as documented. 

6.5
https://docs.vmware.com/en/VMware-vSphere/6.5/com.vmware.psc.doc/GUID-E1D35792-ED03-468A-966B-362BED18021A.html 

6.0
https://docs.vmware.com/en/VMware-vSphere/6.0/com.vmware.vsphere.security.doc/GUID-E1D35792-ED03-468A-966B-362BED18021A.html

Honestly I found several blog posts that instruct to use a chain cert for the machine_ssl cert. This will cause the plugins to fail. What makes this hard is the certificate replacement is reported successful and nothing looks wrong at first. Could be a long time before the impact is noticed. Particularly with a vCenter that doesn't have additional plugins looked at regularly.

Sidenote: Correcting the chain issues in the certificate stores will likely cause Chrome to stop trusting the webclient. Chrome uses the  experimental IETF open standard called Certificate Transparency. https://www.certificate-transparency.org/ vSphere as far as I have seen does not follow this standard. 

This can disabled on a per URL bases in the Windows registry. The key location and additional methods can be found that the link below.
http://www.chromium.org/administrators/policy-list-3#CertificateTransparencyEnforcementDisabledForUrls

**Comment by dbarkelew** [comment]
I wrote feedback on this yesterday, but I guess I missed the submit button?

@jooskim I have found that the JKS is overwritten with a copy of VKS every time the webclient services startup from the tests I have done. The intentional compromised keystore should be self healing to become what is in the VECS stores. Also worth noting every deployment I have fixed with this issue also had issue with certificate replacement originally. However the details were not specified since it wasn't until installing VIC months later that a problem is was noticed in most cases. 

Unless there is a reason for using the redirect the file output switches would match the command documents for documenting the steps. 
`/usr/lib/vmware-vmafd/bin/vecs-cli entry getcert --store MACHINE_SSL_CERT --alias __MACHINE_CERT --output machine_cert.pem`
`keytool -exportcert -keystore /etc/vmware/vsphere-ui/keystore.jks -v -alias __machine_cert -rfc -file virgo_keystore_machine_cert.pem`
Actually in light that the VECS is what creates the JKS step 2 could be omitted to save time, or saved as optional just incase. Unless @jooskim you found that in your case restarting services or rebooting the VC did not fix the compromised keystore.

I think it is worth mentioning for steps 3ii and 4ii that each line should be formatted 64 characters in length or shorter in between the head and footer line. Otherwise openssl, last I checked, will result in an error.

The KB https://kb.vmware.com/s/article/2118939 should be removed. The important steps are better explained by the other KBs listed. Generally there should not be a need to regenerate certificates unless there is a public private key mismatch or expired certificate.

Also need to correct "run the following commands to verify if you get the same SSL public key:" to "run the following commands to verify if you get the same SSL fingerprint:".

**Comment by stuclem** [comment]
@dbarkelew and @jooskim while you are still thrashing out the details, I have revised the workaround in the release notes to just say that we are working on a KB about this issue....

**Comment by jooskim** [comment]
@stuclem works for me. yes this issue is definitely too big for a release note and needs to be a KB article

**Comment by jooskim** [comment]
Adding original bugzilla ticket for reference: https://bugzilla.eng.vmware.com/show_bug.cgi?id=2041514

**Comment by jooskim** [comment]
Internal KB article is available at https://ikb.vmware.com/s/article/52540. @dbarkelew do you know when this article would be made public?

**Comment by dbarkelew** [comment]
@jooskim I would like it to be if possible. Emailing on this.

**Comment by PPisz** [comment]
Hey, I created a small blog post about how to solve this [problem](https://lab.piszki.pl/vmware-integrated-containers-1-3-html5-plugin-ssl-hadshake-error-en/).

Regards,
Piotr

**Comment by jooskim** [comment]
Closing the issue as the KB article is now publicly available at https://kb.vmware.com/kb/52540
@stuclem could you make sure to include it in our troubleshooting docs?

**Comment by stuclem** [comment]
Added the link to the known issues in https://github.com/vmware/vic-ui/releases/tag/v1.3.1#knownissues. 

@jooskim do we want this in the core doc too?

**Comment by jooskim** [comment]
@stuclem I believe so, as this issue can happen in various circumstances.

**Comment by spranta1977** [comment]
Hi all, just for your information. https://kb.vmware.com/s/article/52540?lang=en_US will not work, if your PSC is interacting as an VMCA with a custom certificate. In that case you need the whole SSL Chain, otherwise all other stuff like webclient and so on are going to have an invalid certificate an will be blocked by chrome/firefox. VMware support told me there is no solution for that at the moment. so if you have to configure a VCH you have to do it via vic-commands.

**Comment by mirceasandu** [comment]
I had the same issue with vCenter 6.7 Update 1 and VIC 1.4.4 and it's solved after I've replaced vCenter certificates.

Previously I had custom vCenter certificates:
VC-certificate.cer 
---VC-certificate---
---intermediate-CA---
---root-CA---

VC-key.key

signing-certificate.cer
---intermediate-CA---
---root-CA---

After I've read all posts here, I've reseted all certificates using option 8 from certificate-manager, then I replaced Machine SSL certificate with my custom certificate for vCenter using following template:

VC-certificate.cer
---VC-certificate---

VC-key.key

signing-certificate.cer
---intermediate-CA---
---root-CA---

VIC HTML5 client started to work immediately after taking out intermediate-CA and root-CA from vCenter certificate .cer file.

**Comment by dbarkelew** [comment]
> nvalid certificate and will be blocked by chrome/firefox. VMware support told me there is no solution fo

@spranta1977 
Can you add the VMCA certificate the OS trust store? There is normally a store for intermediate CAs for this type of situation. If using Windows you can normally double click on the certificate with the .cer file extension to import the certificate using the Crypto Shell Extensions wizard. If you are needing to do this for multiple computers using a GPO to push the certificates would probably be the easiest. 

Worth noting Chrome has started requiring a chain for the Certificate Transparency project from Google. This can be disabled to use the traditional certificate validation process on a per URL bases if needed. 
http://www.chromium.org/administrators/policy-list-3#CertificateTransparencyEnforcementDisabledForUrls

**Comment by dbarkelew** [comment]
Review of the plugin certificate handling is tracked further here https://github.com/vmware/vic-ui/issues/635
